app/src/main/java/com/glia/exampleapp/Activity.java

```diff
@@ -4,11 +4,18 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 
+import com.glia.androidsdk.Glia;
+import com.glia.widgets.GliaWidgets;
+
 public class Activity extends AppCompatActivity {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity);
+
+        if (Glia.isInitialized()) return;
+
+        GliaWidgets.init(GliaWidgetsConfigManager.obtainConfig(getIntent(), getApplicationContext()));
```

Comment:
Maybe  in this particular case it's better to  write as follows:

`if (!Glia.isInitialized()) {`
      `GliaWidgets.init()`
`}`

This way `GliaWidgets.init()` is directly related to `!Glia.isInitialized()` and nobody will miss this.

And in the case of `(Glia.isInitialized()) return;` there is a chance that the developer who will be adding some code to onCreate() will not notice early return statement.

Up to  you, it's not critical comment.